geps/gep-1742.md

```diff
@@ -301,8 +301,133 @@ Could not find any HTTP specific timeouts. PRs welcomed. 😊
 
 ## API
 
-TBD.
+The above diagrams show that there are many different kinds of configurable timeouts
+supported by Gateway implementations: connect, idle, request, upstream, downstream.
+Although there may be opportunity for the specification of a common API for more of
+them in the future, this GEP will focus on the L7 timeouts in HTTPRoutes that are
+most valuable to clients.
+
+From the above analysis, it appears that, at a minimum, implementations are capable of
+supporting the configuration of simple client downstream request timeouts on HTTPRoute
+rules. This is a relatively small addition that would benefit many users.
+
+A client HTTP request to a gateway may result in more than one call to the destination
+backend service, for example, if automatic retries are supported. Some implementations
+support configuring a timeout for the individual backend requests, seperate from the
+overall client request timeout. Adding non-Core support for this would also benefit
+many users.
+
+### GO
+
+```go
+type HTTPRouteRule struct {
+	// Timeouts defines the timeouts that can be configured for an HTTP request.
+	//
+	// Support: Core
+	//
+	// +optional
+	// <gateway:experimental>
+	Timeouts *HTTPRouteTimeouts `json:"timeouts,omitempty"`
+
+	// ...
+}
+
+// HTTPRouteTimeouts defines timeouts that can be configured for an HTTPRoute.
+// Timeout values are formatted like 1h/1m/1s/1ms as parsed by Golang time.ParseDuration
+// and MUST BE >= 1ms.
+type HTTPRouteTimeouts struct {
+	// Request specifies a timeout for responding to client HTTP requests,
+	// disabled by default.
+	//
+	// For example, the following rule will timeout if a client request is taking
+	// longer than 10 seconds to complete:
+	//
+	// ```
+	// rules:
+	// - timeouts:
+	//     request: 10s
+	//   backendRefs:
+	//     ...
+	// ```
+	//
+	// Support: Core
+	//
+	// +optional
+	Request *metav1.Duration `json:"request,omitempty"`
+
+	// BackendRequest specifies a timeout for an individual request from the gateway
+	// to a backend service.
+	// Default is the value of Request timeout.
+	//
+	// Support: Extended
+	//
+	// +optional
+	BackendRequest *metav1.Duration `json:"backendRequest,omitempty"`
+}
+```
+
+### YAML
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: timeout-example
+spec:
+  ...
+  rules:
+  - backendRefs:
+    - name: some-service
+      port: 8080
+    timeouts:
+      request: 10s
```

Comment:
This is not `ISO 8601` as specified

Reply:
Changed to say format is as time.ParseDuration

Reply:
thats probably my fault, I added a link to the POC PR mentioning ISO 8601 but didnt read much further into the actual format

apimachinery metav1.Duration is pretty light on documentation too, can maybe just rip the field documentation from go if we need to be specific: https://pkg.go.dev/maze.io/x/duration#ParseDuration